File: _annexe4/annexe4/app/src/main/java/com/example/annexe4/MainActivity.java
package com.example.annexe4;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

public class MainActivity extends AppCompatActivity {


    TextView salutation;
    Utilisateur user;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });


        salutation = findViewById(R.id.text_main);

        //regarde s'il est serialiser
        try {
            FileInputStream fis = openFileInput("fichier.ser");
            ObjectInputStream ois = new ObjectInputStream(fis);
            user = (Utilisateur) ois.readObject();
            salutation.setText( "Bonjour " + user.prenom + " " + user.nom  );
        } catch (IOException | ClassNotFoundException ignored) {

        }

        //sinon regarde s'il est dans le savedInstance Bundle
        if(user == null){
            try {
                user = (Utilisateur) savedInstanceState.getSerializable("Utilisateur");
                salutation.setText( "Bonjour " + user.prenom + " " + user.nom  );
            } catch (Exception ignored) {

            }
        }

        ActivityResultLauncher<Intent> lanceur;
        lanceur = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(), new CallBackInfo());

        Button next = findViewById(R.id.btn_main);

        next.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(),infoActivity.class);
                lanceur.launch(intent);
            }
        });

    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        if(user != null)
            outState.putSerializable("Utilisateur", user);
        try {
            FileOutputStream fos =  openFileOutput("fichier.ser", MODE_PRIVATE);
            ObjectOutputStream oos = new ObjectOutputStream(fos);
            oos.writeObject(user);

        } catch (IOException ignored) {

        }
    }

    public class CallBackInfo implements ActivityResultCallback<ActivityResult> {
        // appelé quand je reviens  dans cette activité, retour du boomerang
        @Override
        public void onActivityResult(ActivityResult result) {
            if (result.getResultCode() == RESULT_OK) {
                user = (Utilisateur) result.getData().getSerializableExtra("Utilisateur");
                salutation.setText( "Bonjour " + user.prenom + " " + user.nom  );
            }

        }
    }



}

File: _annexe3 (singleton)/annexe3c/projetDentiste/app/src/main/java/com/factures/applicationpaiementfactures/MainActivity.kt
package com.factures.applicationpaiementfactures

import android.os.Bundle
import android.os.PersistableBundle
import android.widget.CheckBox
import android.widget.EditText
import android.widget.LinearLayout
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.io.IOException
import java.io.ObjectInputStream
import java.io.ObjectOutputStream


class MainActivity : AppCompatActivity() {

    lateinit var dent1: LinearLayout
    lateinit var dent2: LinearLayout


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }


        /*
        dent1 = findViewById(R.id.dent1)
        var num1:String
        var canal1:Boolean
        var notes1:String

        dent2 = findViewById(R.id.dent2)
        var num2:String
        var canal2:Boolean
        var notes2:String



        try {
            val fis = openFileInput("fichier.ser")
            val ois = ObjectInputStream(fis)

            println("fichier trouvé")
            num1 = ois.readObject() as String
            canal1 = ois.readBoolean()
            notes1 = ois.readObject() as String

            num2 = ois.readObject() as String
            canal2 = ois.readBoolean()
            notes2 = ois.readObject() as String

            (dent1.getChildAt(0) as EditText).setText(num1)
            (dent1.getChildAt(1) as CheckBox).isChecked = canal1
            (dent1.getChildAt(2) as EditText).setText(notes1)

            (dent2.getChildAt(0) as EditText).setText(num2)
            (dent2.getChildAt(1) as CheckBox).isChecked = canal2
            (dent2.getChildAt(2) as EditText).setText(notes2)


            println("terminé")
        } catch (ignored: IOException) {
        } catch (ignored: ClassNotFoundException) {
        }

         */



    }

    override fun onCreate(savedInstanceState: Bundle?, persistentState: PersistableBundle?) {
        super.onCreate(savedInstanceState, persistentState)
        println("onCreate")
    }

    override fun onStart() {
        super.onStart()
        println("onStart")
    }

    override fun onResume() {
        super.onResume()
        println("onResume")
    }

    override fun onRestart() {
        super.onRestart()
        println("onRestart")
    }

    override fun onPause() {
        super.onPause()
        println("onPause")
    }

    override fun onDestroy() {
        super.onDestroy()
        println("onDestroy")
    }


    override fun onStop() {
        super.onStop()
        println("onStop")
        /*
        try {
            val fos = openFileOutput("fichier.ser", MODE_PRIVATE)
            val oos = ObjectOutputStream(fos)
            oos.writeObject(  ((dent1.getChildAt(0) as EditText).text.toString()) )
            oos.writeBoolean( ((dent1.getChildAt(1) as CheckBox).isChecked)  )
            oos.writeObject( ((dent1.getChildAt(2) as EditText).text.toString())   )


            oos.writeObject(  ((dent2.getChildAt(0) as EditText).text.toString()) )
            oos.writeBoolean( ((dent2.getChildAt(1) as CheckBox).isChecked)  )
            oos.writeObject( ((dent2.getChildAt(2) as EditText).text.toString())   )
        } catch (ignored: IOException) {
        }

         */

    }
}